src/main/java/co/rsk/federate/signing/hsm/config/PowHSMConfig.java

```diff
@@ -0,0 +1,153 @@
+package co.rsk.federate.signing.hsm.config;
+
+import static co.rsk.federate.signing.hsm.config.PowHSMConfigParameter.*;
+
+import co.rsk.bitcoinj.core.NetworkParameters;
+import co.rsk.federate.signing.config.SignerConfig;
+import co.rsk.federate.signing.hsm.HSMClientException;
+import co.rsk.federate.signing.hsm.HSMUnsupportedTypeException;
+import co.rsk.federate.signing.hsm.client.HSMBookkeepingClient;
+import com.typesafe.config.Config;
+import java.math.BigInteger;
+import java.util.function.Function;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class PowHSMConfig {
+
+  private static final Logger logger = LoggerFactory.getLogger(PowHSMConfig.class);
+  private static final String SIGNER_TYPE = "hsm";
+
+  private final Config config;
+
+  private PowHSMConfig(Config config) {
+    this.config = config;
+  }
+
+  public static PowHSMConfig from(SignerConfig signerConfig) {
+    if (signerConfig == null || !signerConfig.getType().equals(SIGNER_TYPE)) {
+      return null;
```

Comment:
The way it was handled before is that it returned null. But we can change it to throw an exception, what exception do you have in mind?

Reply:
I prefer to handle it with an exception yes. Probably some type of HSMClientException makes sense. I think the existing ones are all related to HSM response codes. Perhaps we should create a new one. Or maybe even use `SignerException`

What do you think?

Reply:
Also, have a close look on how we will handle this exception.

Does it mean that the node can not start or simply that it is not using a PowHSM signer?

Reply:
Sounds good, I decided to use `SIgnerException` since it already being used for similar reasons. The new changes I have included lets the node start if there is no powHSM config found / not using a PowHSM signer (it will only log about it). Let me know if this looks better.